payments/providers/cpu_ceepos.py

```diff
@@ -0,0 +1,360 @@
+import hashlib
+import hmac
+import logging
+
+import requests
+from django.http import HttpResponse
+from requests.exceptions import RequestException
+
+from ..exceptions import (
+    DuplicateOrderError,
+    OrderStateTransitionError,
+    PayloadValidationError,
+    PaymentCreationFailedError,
+    ServiceUnavailableError,
+    UnknownReturnCodeError,
+)
+from ..models import Order, OrderLine
+from ..utils import price_as_sub_units
+from .base import PaymentProvider
+
+logger = logging.getLogger(__name__)
+
+# Keys the provider expects to find in the config
+RESPA_PAYMENTS_CEEPOS_API_URL = "RESPA_PAYMENTS_CEEPOS_API_URL"
+RESPA_PAYMENTS_CEEPOS_API_KEY = "RESPA_PAYMENTS_CEEPOS_API_KEY"
+RESPA_PAYMENTS_CEEPOS_API_SECRET = "RESPA_PAYMENTS_CEEPOS_API_SECRET"
+
+
+class CPUCeeposProvider(PaymentProvider):
+    """CPU Ceepos specific integration utilities and configuration"""
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.url_payment_api = self.config.get(RESPA_PAYMENTS_CEEPOS_API_URL)
+
+    @staticmethod
+    def get_config_template() -> dict:
+        """Keys and values that CeePos requires from environment"""
+        return {
+            RESPA_PAYMENTS_CEEPOS_API_URL: str,
+            RESPA_PAYMENTS_CEEPOS_API_KEY: str,
+            RESPA_PAYMENTS_CEEPOS_API_SECRET: str,
+        }
```

Comment:
Typing a dict return value should be done with `typing.Dict` and specifying the key and value types too:
```python
from typing import Dict

def get_config_template() -> Dict[str, str]:
    return { ...
```

Or if you raise the Python requirement up to 3.9 or newer, you can also use `dict[str, str]` without the import. See https://www.python.org/dev/peps/pep-0585/

Reply:
Thanks, I adjusted this 👍